File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    
    for(int i=0; i<n; i++){
        string res = "";
        arr1[i] = arr1[i] | arr2[i];
        
        for(int j=0; j<n;j++){
            if(arr1[i]%2==0) res = " " + res;
            else    res = "#" + res;
            arr1[i] = arr1[i] >> 1;
        }
        
        answer.push_back(res);
    }
    
    return answer;
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = [];
    
    for(let i=0; i<n; i++){
        const rawBinary = (arr1[i] | arr2[i]).toString(2);
        const binary = rawBinary.padStart(n, '0');
        answer.push(binary);
    }
    
    for(let i=0; i<n; i++){
        const arr = answer[i].split("1");
        const s = arr.join("#");
        const arr2 = s.split("0");
        const s2 = arr2.join(" ");
        answer[i] = s2;
    }
    

 
 
    
    return answer;
}